src/App.tsx

```diff
@@ -14,11 +15,96 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGoods: string[];
+};
 
-export default App;
+export class App extends React.Component<{}, State> {
+  state: State = {
+    selectedGoods: [],
+  };
+
+  addItem = (item: string) => {
+    const { selectedGoods } = this.state;
+
+    if (!selectedGoods.includes(item)) {
+      this.setState({ selectedGoods: [...selectedGoods, item] });
+
+      return;
+    }
+
+    this.setState({
+      selectedGoods: selectedGoods.filter(inCart => inCart !== item),
+    });
+  };
```

Comment:
I like that you've made one function for adding and deleting,  👍  but perhaps it needs a better name in this case. smth like addDelete or whatever

Reply:
oh yes, i renamed it in the local file but forgot to push it😄